File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/resource/KotlinIosResources.java
package ru.pocketbyte.locolaser.platform.kotlinmobile.resource;

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.file.KotlinIosResourceFile;
import ru.pocketbyte.locolaser.resource.file.ResourceFile;

import java.io.File;
import java.util.Set;

public class KotlinIosResources extends AbsKotlinImplementationPlatformResources {

    public KotlinIosResources(File dir, String name, String interfaceName) {
        super(dir, name, interfaceName);
    }

    @Override
    protected ResourceFile[] getResourceFiles(Set<String> locales) {
        return new ResourceFile[] {
                new KotlinIosResourceFile(getFile(),
                        this.className, this.classPackage,
                        this.interfaceName, this.interfacePackage)
        };
    }
}

File: platform-kotlin-mpp/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmpp/resource/KotlinIosResources.kt
package ru.pocketbyte.locolaser.platform.kotlinmpp.resource

import ru.pocketbyte.locolaser.platform.kotlinmpp.resource.file.KotlinIosResourceFile
import ru.pocketbyte.locolaser.resource.file.ResourceFile
import ru.pocketbyte.locolaser.resource.formatting.FormattingType
import ru.pocketbyte.locolaser.resource.formatting.JavaFormattingType

import java.io.File

class KotlinIosResources(
        dir: File,
        name: String,
        interfaceName: String,
        filter: ((key: String) -> Boolean)?
) : AbsKotlinImplementationPlatformResources(dir, name, interfaceName, filter) {

    override val formattingType: FormattingType = JavaFormattingType

    override fun getResourceFiles(locales: Set<String>?): Array<ResourceFile>? {
        return arrayOf(KotlinIosResourceFile(directory,
                this.className, this.classPackage,
                this.interfaceName, this.interfacePackage))
    }
}
